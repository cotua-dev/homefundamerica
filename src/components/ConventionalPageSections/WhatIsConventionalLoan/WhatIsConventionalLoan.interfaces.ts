export interface FetchWhatIsConventionalLoanImageQuery {
    whatIsConventionalLoanImage: {
        publicURL: string;
    };
}
