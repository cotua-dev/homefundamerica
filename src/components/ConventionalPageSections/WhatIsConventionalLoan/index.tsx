import React, {
    FunctionComponent,
    ReactElement,
} from 'react';
import {
    useStaticQuery,
    graphql,
} from 'gatsby';
import { moveToStepper } from '../../../utils/moveToStepper';
import { whatIsConventionalLoanSection } from './WhatIsConventionalLoan.module.scss';
import * as sharedStyles from '../../sharedStyles/shared.module.scss';

export const WhatIsConventionalLoan: FunctionComponent = (): ReactElement => {
    const queryResult = useStaticQuery(graphql`
        query {
            whatIsConventionalLoanImage: file(name: {eq: "AdobeStock_327425317"}, extension: {eq: "jpg"}) {
                publicURL
            }
        }
    `);

    return (
        <section id={whatIsConventionalLoanSection} className={sharedStyles.loanPositive}>
            <div className={sharedStyles.contentWrapper}>
                <h2>{`What is a Conventional Loan?`}</h2>
                <p>
                    {`
                        A conventional loan is a mortgage that follows the guidelines
                        set by Fannie Mae and Freddie Mac. These government-sponsored
                        companies help provide lenders with the money they need to
                        make home loans. Loan size, limited to $484,350 in most U.S.
                        counties, is the most well-known requirement for conventional
                        loans. Loans that don't meet the standards set are called
                        Non-Conventional Loans.
                    `}
                </p>
                <button
                    className={sharedStyles.applyButton}
                    type="button"
                    onClick={moveToStepper}
                >Apply Now</button>
            </div>
            <div className={sharedStyles.imageWrapper}>
                <img
                    src={queryResult?.whatIsConventionalLoanImage.publicURL}
                    alt="What is a conventional loan image"
                />
            </div>
        </section>
    );
}
