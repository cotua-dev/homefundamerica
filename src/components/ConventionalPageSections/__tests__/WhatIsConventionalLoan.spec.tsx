import React from 'react';
import {
    render,
    RenderResult,
} from '@testing-library/react';
import { WhatIsConventionalLoan } from '../WhatIsConventionalLoan';

describe('WhatIsConventionalLoan Component', (): void => {
    test('renders', (): void => {
        const { container }: RenderResult = render(<WhatIsConventionalLoan/>);
        expect(container.firstChild).toMatchSnapshot();
    });
});
