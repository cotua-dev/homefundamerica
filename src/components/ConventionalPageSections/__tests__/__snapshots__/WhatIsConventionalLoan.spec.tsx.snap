// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WhatIsConventionalLoan Component renders 1`] = `
<section
  id="whatIsConventionalLoanSection"
>
  <div>
    <h2>
      What is a Conventional Loan?
    </h2>
    <p>
      
                        A conventional loan is a mortgage that follows the guidelines
                        set by Fannie Mae and Freddie Mac. These government-sponsored
                        companies help provide lenders with the money they need to
                        make home loans. Loan size, limited to $484,350 in most U.S.
                        counties, is the most well-known requirement for conventional
                        loans. Loans that don't meet the standards set are called
                        Non-Conventional Loans.
                    
    </p>
    <button
      type="button"
    >
      Apply Now
    </button>
  </div>
  <div>
    <img
      alt="What is a conventional loan image"
    />
  </div>
</section>
`;
